desk_test.go

```diff
@@ -106,23 +123,83 @@ func (tsp testScreensProvider) ScreenForWindow(win Window) *Screen {
 	return tsp.Screens()[0]
 }
 
+type testAppData struct {
+	name string
+}
+
+func (tad *testAppData) Name() string {
+	return tad.name
+}
+
+func (tad *testAppData) Run([]string) error {
+	return nil
+}
+
+func (tad *testAppData) Icon(theme string, size int) fyne.Resource {
+	if theme == "" {
+		return nil
+	} else if theme == "Maximize" {
+		return wmTheme.MaximizeIcon
+	}
+	return wmTheme.IconifyIcon
+}
+
+type testAppProvider struct {
+	screens []*Screen
+}
+
+func (tap *testAppProvider) AvailableApps() []AppData {
+	return nil
+}
+
+func (tap *testAppProvider) AvailableThemes() []string {
+	return nil
+}
+
+func (tap *testAppProvider) FindAppFromName(appName string) AppData {
+	return &testAppData{name: appName}
+}
+
+func (tap *testAppProvider) FindAppFromWinInfo(win Window) AppData {
+	return &testAppData{}
+}
+
+func (tap *testAppProvider) FindAppsMatching(pattern string) []AppData {
+	return nil
+}
+
+func (tap *testAppProvider) DefaultApps() []AppData {
+	return nil
+}
+
+func newTestAppProvider() ApplicationProvider {
+	return &testAppProvider{}
+}
+
 func newTestScreensProvider() ScreenList {
 	return &testScreensProvider{}
 }
 
-func TestDeskLayout_Layout(t *testing.T) {
+func NewTestDesktop() Desktop {
```

Comment:
No need to export a test constructor.
You'd avoid a lot of casting if this returned *deskLayout.